rasterio/_env.pyx

```diff
@@ -56,12 +56,11 @@ cdef bint is_64bit = sys.maxsize > 2 ** 32
 cdef void log_error(CPLErr err_class, int err_no, const char* msg) with gil:
     """Send CPL debug messages and warnings to Python's logger."""
     log = logging.getLogger('rasterio._gdal')
-    if err_no in code_map:
-        # 'rasterio._gdal' is the name in our logging hierarchy for
-        # messages coming direct from CPLError().
-        log.log(level_map[err_class], "%s in %s", code_map[err_no], msg)
-    else:
-        log.info("Unknown error number %r", err_no)
+    if err_class < 3:
```

Comment:
Before: we logged *all* GDAL errors, even the ones we were *also* turning into Python exceptions. Even if we handled the exception, the log message would still be emitted, which confused the heck out of me recently.

Reply:
Might be good to add brief instructions on how to log errors / exceptions properly from GDAL in our Wiki: https://github.com/mapbox/rasterio/wiki/Cython-and-GDAL